src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ExplicitInitializationCheck.java

```diff
@@ -181,8 +181,7 @@ private void validateNonObjects(DetailAST ast) {
      */
     private static boolean isZeroChar(DetailAST exprStart) {
         return isZero(exprStart)
-            || exprStart.getType() == TokenTypes.CHAR_LITERAL
-            && "'\\0'".equals(exprStart.getText());
+            || "'\\0'".equals(exprStart.getText());
```

Comment:
Don't see how to satisfy `"'\\0'".equals(exprStart.getText()) && exprStart.getType() != TokenTypes.CHAR_LITERAL`.